### PR TITLE
View cycles

### PR DESCRIPTION
**Pull request by dwagg**

### CODE REVIEW

**Review by whuiras — APPROVED** [review]

**Review by JasonRitz — APPROVED** [review]
All looks good. Tests pass and the accordion view works as expected.

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestCycleListModel.swift [hunk 1]
```diff
@@ -11,9 +11,10 @@ import Foundation
 class TestCycleListModel {
     var testCycleList: [TestCycleModel] = []
     
-    func extractCycleList(fromData: [[String : AnyObject]]) {
+    func extractCycleList(fromData: [[String : AnyObject]], parentId: Int) {
         for cycle in fromData {
-            if cycle["itemType"] as! Int != 36 {
+            let fields: [String : AnyObject] = cycle["fields"] as! Dictionary
+            if cycle["itemType"] as! Int != 36 || fields["testPlan"] as! Int != parentId{
```
Comment (lcc4) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestCycleListModel.swift]: @dwagg can you clarify why this has a hardcoded 36?

**Review by lcc4 — CHANGES_REQUESTED** [review]

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestListViewController.swift [hunk 2]
```diff
@@ -49,9 +52,6 @@ class TestListViewController: UIViewController {
     
     // TO DO: attach this action to the selected plan button
```
Comment (lcc4) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestListViewController.swift]: Is this comment valid or out-of-date?

File: Mobile Test Runner for Jama/Mobile Test Runner for JamaTests/TestCycleListModelUnitTests.swift [hunk 3]
```diff
@@ -21,6 +21,7 @@ class TestCycleListModelUnitTests: XCTestCase {
     var projectId = 123
     var cycle1Name = "cycle1"
     var cycle2Name = "cycle2"
+    var testPlanId = 12
     var testCycleItemType = 36
```
Comment (lcc4) [inline Mobile Test Runner for Jama/Mobile Test Runner for JamaTests/TestCycleListModelUnitTests.swift:25]: Is this why the hardcoded 36 above? To differentiate if it is a testCycle or not testCycle? Where did those numbers come from?

Comment (dwagg) [reply Mobile Test Runner for Jama/Mobile Test Runner for JamaTests/TestCycleListModelUnitTests.swift:25]: Yes I refactored it to be a constant that has a inline comment explaining the numbers. It is the Jama system ID for test cycles.

File: Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestListViewController.swift [hunk 4]
```diff
@@ -97,14 +95,15 @@ extension TestListViewController: EndpointDelegate {
...
-                    tmpList.extractCycleList(fromData: unwrappedData)
+                    tmpList.extractCycleList(fromData: unwrappedData, parentId: self.planId)
                     if tmpList.testCycleList.isEmpty {
                         return
                     }
                     self.testCycleList.testCycleList.append(contentsOf: tmpList.testCycleList)
-
+                    self.totalCyclesVisable = self.testCycleList.testCycleList.count
+                    self.testPlanList.testPlanList[self.selectedPlanIndex].numOfCycles = self.totalCyclesVisable
                     //reload Data in view? self.testCycle.reloadData()
```
Comment (lcc4) [inline Mobile Test Runner for Jama/Mobile Test Runner for Jama/TestListViewController.swift]: what is this comment signaling? Is this work that needs to be done or is it out-of-date?

**Review by lcc4 — APPROVED** [review]

**Pull request merged**